core/internal/pathtree/pathtree.go

```diff
@@ -34,8 +40,25 @@ type Format int
 const (
 	FormatYaml Format = iota
 	FormatJson
+	FormatJsonExt
 )
 
+// PathItem is a alternative representation of the item interface.
+//
+// The Value is a JSON string, which can be unmarshaled to any type.
+type PathItem struct {
+	Path  TreePath
+	Value string
+}
+
+// Leaf is a the leaf node in the tree.
+//
+// The Value could be any primitive type, list.
+type Leaf struct {
+	Path  TreePath
+	Value any
+}
```

Comment:
it seems this is redundant with PathItem

Reply:
Yeah, can be switch to using type verification. That’s an option

Reply:
I mean the structures are almost identical. If you remove JSON handling from PathTree, then `PathItem.Value` can be `any` or a generic type and you don't need `Leaf`.

Reply:
i understand, it's just i'm not sure it's a good idea to remove the json handling.
unless you have a way to avoid redundant tree traversal